### PR TITLE
Fixed bug causing performance deterioration

### PR DESCRIPTION
**Pull request by novichkovg**
Fixed bug causing incorrect `null_count` computation as well as severe performance degradation for large fragment size.

### CODE REVIEW

File: QueryEngine/ArrowResultSetConverter.cpp [hunk 1]
```diff
@@ -311,49 +311,54 @@ void convert_column(ResultSetPtr result,
...
         threading::blocked_range<size_t>(static_cast<size_t>(0), unroll_count / 8),
         [&](auto r) {
+          int64_t local_null_count = 0;
           for (auto i = r.begin() * 8; i < r.end() * 8; i += 8) {
             uint8_t valid_byte = 0;
             uint8_t valid;
             valid = vals[i + 0] != null_val;
             valid_byte |= valid << 0;
-            null_count += !valid;
+            local_null_count += !valid;
```
Comment (fexolm) [inline QueryEngine/ArrowResultSetConverter.cpp:325]: Any reason, why we can not just use loop over bitmap offset here?

**Review by fexolm — APPROVED** [review]
LGTM, just wondering, why do we need to manually unroll the loop here.

**Comment by ienkovich** [comment]
> LGTM, just wondering, why do we need to manually unroll the loop here.

Inner loop with 8 iterations would most probably be unrolled by the compiler.

**Review by ienkovich — APPROVED** [review]

File: QueryEngine/ArrowResultSetConverter.cpp (hunk 1 above)
Comment (novichkovg) [reply QueryEngine/ArrowResultSetConverter.cpp:325]: > Any reason, why we can not just use loop over bitmap offset here?

I thought about that too, but then one would probably have to unroll it to make it faster.

**Pull request merged**